go-controller/pkg/ovn/master.go

```diff
@@ -765,20 +769,50 @@ func (oc *DefaultNetworkController) addUpdateLocalNodeEvent(node *kapi.Node, nSy
 		}
 	}
 
+	if nSyncs.syncZoneIc && config.OVNKubernetesFeature.EnableInterconnect {
+		if err := oc.zoneChassisHandler.AddLocalZoneNode(node); err != nil {
+			errs = append(errs, err)
+			oc.syncZoneIcFailed.Store(node.Name, true)
+		} else {
+			if err := oc.zoneICHandler.AddLocalZoneNode(node); err != nil {
+				errs = append(errs, err)
+				oc.syncZoneIcFailed.Store(node.Name, true)
+			} else {
+				oc.syncZoneIcFailed.Delete(node.Name)
+			}
+		}
+	}
+
 	err = kerrors.NewAggregate(errs)
 	if err != nil {
 		oc.recordNodeErrorEvent(node, err)
 	}
+
 	return err
 }
 
-func (oc *DefaultNetworkController) addUpdateRemoteNodeEvent(node *kapi.Node) error {
+func (oc *DefaultNetworkController) addUpdateRemoteNodeEvent(node *kapi.Node, syncZoneIc bool) error {
 	_, present := oc.localZoneNodes.Load(node.Name)
 
 	if present {
 		_ = oc.deleteNodeEvent(node)
 	}
-	return nil
+
+	var err error
+	if syncZoneIc && config.OVNKubernetesFeature.EnableInterconnect {
+		if err = oc.zoneChassisHandler.AddRemoteZoneNode(node); err != nil {
+			klog.Error(err)
+			oc.syncZoneIcFailed.Store(node.Name, true)
+		} else {
+			if err = oc.zoneICHandler.AddRemoteZoneNode(node); err != nil {
+				klog.Error(err)
+				oc.syncZoneIcFailed.Store(node.Name, true)
+			} else {
+				oc.syncZoneIcFailed.Delete(node.Name)
+			}
+		}
+	}
+	return err
```

Comment:
return fmt.Errorf("adding or updating remote node %s failed, err %w", node.Name, err) do something like that..